From fdaa3d9c5286d41d943633bedeafebc2df7d2fa3 Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Fri, 14 Nov 2014 14:22:10 +0100
Subject: [PATCH] net2280: Fix superspeed request completion

Parameter three in function call dma_done() is incorrect. Read value of
'dmacount' again.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 drivers/usb/gadget/udc/net2280.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/net2280.c b/drivers/usb/gadget/udc/net2280.c
index 2e95715..2e83f09 100644
--- a/drivers/usb/gadget/udc/net2280.c
+++ b/drivers/usb/gadget/udc/net2280.c
@@ -1120,7 +1120,7 @@ static void scan_dma_completions(struct net2280_ep *ep)
 				(req->req.length % ep->ep.maxpacket) != 0) {
 			tmp = readl(&ep->regs->ep_stat);
 			if (ep->dev->quirks & PLX_SUPERSPEED)
-				return dma_done(ep, req, tmp, 0);
+				return dma_done(ep, req, readl(&ep->dma->dmacount), 0);
 
 			/* AVOID TROUBLE HERE by not issuing short reads from
 			 * your gadget driver.  That helps avoids errata 0121,
-- 
1.8.4.5

